File: frontend/src/types/account.ts
export interface Card {
  type: "Virtual" | "Physical";
  last4: string;
  status: "Active" | "Frozen" | "Cancelled";
}

export interface Account {
  id: string;
  name: string;
  email: string;
  accountNumber: string;
  tier: "Premium" | "Standard";
  balance: number;
  currency: string;
  monthlyChange: number;
  joinDate: string;
  cards: Card[];
}

File: backend/app/models/account.py
from pydantic import BaseModel


class CardResponse(BaseModel):
    type: str
    last4: str
    status: str


class AccountResponse(BaseModel):
    id: str
    name: str
    email: str
    account_number: str
    tier: str
    balance: float
    currency: str
    monthly_change: float
    join_date: str
    cards: list[CardResponse]
